### PR TITLE
Single post

### PR DESCRIPTION
**Pull request by pariyar07**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
       <Routes>
-        <Route path="/" element={<Homepage />} />
-        <Route path="*" element={<div style={{ display: "flex", justifyContent: "center" }}><h1 style={{ fontSize: "5rem" }}>Page Not Found!</h1></div>} />
+        <Route path="/login" element={<Login />} />
+        <Route path="/signup" element={<SignUp />} />
+        <Route path="/" element={<RequiresAuth><Homepage /></RequiresAuth>} />
+        <Route path="/bookmarks" element={<RequiresAuth><Bookmark /></RequiresAuth>} />
+        <Route path="/post/:postId" element={<RequiresAuth><SinglePost /></RequiresAuth>} />
+        <Route path="/profile" element={<RequiresAuth><Profile/></RequiresAuth>} />
+        <Route path="*" element={<div className="flex justify-center items-center min-w-full"><h1 style={{ fontSize: "5rem" }}>Page Not Found!</h1></div>} />
```
Comment (shivanigangadharan) [inline src/App.js:20]: suggestion to make a separate component here for page not found

**Review by shivanigangadharan — APPROVED** [review]
Code looks well structured and clean, can merge this.

File: src/frontend/pages/homepage/homepage.jsx [hunk 2]
```diff
@@ -1,33 +1,112 @@
...
+            scale: {
+                scaleCount: 0,
+                scaledBy: []
+            },
+            comments: [],
+        } : {
+            _id: uuid(),
+            content: content,
+            image: URL.createObjectURL(image),
+            username: "Admin",
```
Comment (vishalg8454) [inline src/frontend/pages/homepage/homepage.jsx:57]: No need to send all the fields in while making a post. The backend will take care of adding all attributes. You can simply send the two strings. One for post text and another for image.

**Review by vishalg8454 — COMMENTED** [review]
Good to merge. Left a comment, do check that.

**Comment by vishalg8454** [comment]
Also the Navbar and left navigation should stick instead of being scrolled.

**Pull request merged**